### PR TITLE
class to calculate fit fractions

### PR DESCRIPTION
**Pull request by admorris**
* Moved the fit fraction calculation from `main.cpp` to its own class: `FitFractionCalculator`.
* Handles multiple discrete combinations.
* Can print to terminal.
* Can write results to a file.
  * One tree per discrete combination.
  * One branch per component.
  * Add a new entry to the tree if it already exists (useful for toy studies).

### CODE REVIEW

File: framework/src/FitFractionCalculator.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	{
+		std::cout << "Fit fractions in combination " << combination.first << "\n";
+		for(const auto& fraction: combination.second)
+			std::cout << std::setw(30) << fraction.first << " : " << fraction.second << "\n";
+		std::cout << std::endl;
+	}
+}
+void FitFractionCalculator::WriteToFile(std::string filename)
+{
+	auto OutputFile = std::unique_ptr<TFile>(TFile::Open(filename.c_str(),"UPDATE"));
```
Comment (rob-c) [inline framework/src/FitFractionCalculator.cpp:43]: Not to dissuade the use of things like `unique_ptr` or `auto` why isn't this just `TFile* OutputFile = TFile::Open(...` ?

Comment (admorris) [reply framework/src/FitFractionCalculator.cpp:43]: The documentation for `TFile::Open` says "the caller is responsible for deleting the pointer", so either I use a smart pointer or write a `delete` statement at the end of the scope. Given the choice of those two, I'd rather go with the one that doesn't involve manual memory management.

Comment (rob-c) [reply framework/src/FitFractionCalculator.cpp:43]: OK, was just curious. Thanks for the explanation.
I'm fairly old school in that I wrote most of my c++ way before the 11 standard was implemented :stuck_out_tongue:
(Clearly I need to brush up on my modern c++)

**Comment by gcowan** [comment]
Thanks for this - nice to have this put into its own class and not the hack that I had.

**Pull request merged**